posthog-nextjs/src/webpack-plugin.ts

```diff
@@ -0,0 +1,93 @@
+import { PostHogNextConfigComplete } from './config'
+import { spawn } from 'child_process'
+
+type NextRuntime = 'edge' | 'nodejs' | undefined
+
+export class SourcemapWebpackPlugin {
```

Comment:
Maybe something that can come afterwards but we should abstract this to be a generic plugin any webpacker can use